### DISCUSSION TITLE
How to do multitouch gestures: pinch/scale rotate?

### DISCUSSION THREAD

**Discussion by TimKraemer** [pixijs/pixijs#9307] [General]
Hi everyone!

I'm new to pixijs (actually using [@pixi/react](https://github.com/pixijs/pixi-react)) and I'm curious what the appropriate way to use multi-touch gestures is for pixi.
I understand that this library is made for rendering 2D primarily, so I was thinking about using another library to handle touches and interpret gestures for me (e.g. https://github.com/pmndrs/use-gesture), but I noticed that it doesn't "just" work, most likely because of those libraries expect different event handlers like `onMouseDown` and `onTouchStart`.
But again, I'm very new at this, so I might be missing something obvious.... however I didn't find anything at all in the docs or here in github which is somewhat recent and related to multitouch gestures....

For now I work around the issue with [one-finger dragging](https://pixijs.io/examples/#/events/dragging.js) and [pixi transformer](https://api.pixijs.io/@pixi-essentials/transformer/Transformer.html). Obviously an ugly solution for mobile though :/

Any ideas/examples on what to combine to let me use the advantages of pixi and also the user-experience of multi-touch gestures? E.g. an example on how to drag/rotate/scale a sprite?
Thank you!

**Comment by tommie** [comment]
It looks like Pixi's touch events aren't really touch events. If the device supports "pointer" events, it takes precedence

https://github.com/pixijs/pixijs/blob/c04c09c50874fef9296555af23e25fe58cc808a9/packages/events/src/EventSystem.ts#L570-L594

Normally, you can get a list of "touches" from events, but it looks like we'll have to implement that ourselves. In my usecase, I just want to P-Z the entire viewport, so the events would be global anyway, and I can just use the canvas element's events. For individual sprites, it would require a bit more work, it seems.
